airflow/jobs/scheduler_job_runner.py

```diff
@@ -277,9 +284,44 @@ def __get_concurrency_maps(self, states: Iterable[TaskInstanceState], session: S
             .where(TI.state.in_(states))
             .group_by(TI.task_id, TI.run_id, TI.dag_id)
         )
-        return ConcurrencyMap.from_concurrency_map(
-            {(dag_id, run_id, task_id): count for task_id, run_id, dag_id, count in ti_concurrency_query}
+        ti_concurrency = {
+            (dag_id, run_id, task_id): count for task_id, run_id, dag_id, count in ti_concurrency_query
+        }
+
+        tg_concurrency_query: list[tuple[str, str, str, str, int]] = (
+            session.query(TI.task_id, TI.run_id, TI.dag_id, TI.map_index)
+            .filter(
+                TI.state.in_(
+                    [
+                        State.SCHEDULED,
+                        State.QUEUED,
+                        State.RUNNING,
+                        State.UP_FOR_RESCHEDULE,
+                        State.UP_FOR_RETRY,
+                    ]
+                )
+            )
+            .filter(TI.map_index >= 0)
+            .order_by(TI.dag_id, TI.run_id, TI.map_index)
         )
+        tg_concurrency: dict[tuple[str, str, str], tuple[int, set[int]]] = dict()
```

Comment:
This type is too complex, add a type alias for it to better explain what the key and value represent.

Reply:
Also, use `{}` instead of `dict()` (there are multiple instances in the PR)

Reply:
> Also, use `{}` instead of `dict()` (there are multiple instances in the PR)

applied to the code.

Reply:
type alias added